src/main.nr

```diff
@@ -8,6 +8,16 @@ fn main(value: u32, hash_value: pub u32){
 
 fn calculate_hash(num: u32) -> u32 {
```

Comment:
Since Noir automatically does mod when the value exceeds the range. You can do something like this to avoid doing mod

```
fn calculate_hash(num: u32) -> u32 {

    let num_bytes = (num as Field).to_be_bytes(4);
    let mut hash_value: u32 = 0x811C9DC5;

    for i in 0..4 {
        let product = hash_value * PRIME;
        hash_value = product ^ (num_bytes[i] as u32);
    }

    hash_value
}
```